rtgui/threadutils.h

```diff
@@ -92,8 +90,8 @@ class MyRWMutex :
     friend class MyWriterLock;
 
 private:
-    Glib::Threads::Mutex mutex;
-    Glib::Threads::Cond cond;
+    std::mutex mutex;
+    std::condition_variable_any cond;
```

Comment:
[cppreference](https://en.cppreference.com/w/cpp/thread/condition_variable_any) states:
> If the lock is std::unique_lock, std::condition_variable may provide better performance.

Maybe something like this?

```patch
diff --git a/rtgui/threadutils.cc b/rtgui/threadutils.cc
index 4f8190c36..b7e18e955 100644
--- a/rtgui/threadutils.cc
+++ b/rtgui/threadutils.cc
@@ -68,7 +68,7 @@ void MyReaderLock::acquire ()
         return;
     }
 
-    std::lock_guard<std::mutex> lock (mutex.mutex);
+    std::unique_lock<std::mutex> lock (mutex.mutex);
 
     if (mutex.writerCount == 0) {
         // There's no writer operating, we can increment the writer count which will lock writers.
@@ -77,7 +77,7 @@ void MyReaderLock::acquire ()
         // The writer count is non null, but a reader can be the owner of the writer lock,
         // which will be the case if the reader count is not zero too.
         while (mutex.writerCount != 0) {
-            mutex.cond.wait(mutex.mutex);
+            mutex.cond.wait(lock);
         }
 
         // Then, we can increment the writer count.
@@ -96,7 +96,7 @@ void MyReaderLock::release ()
         return;
     }
 
-    std::lock_guard<std::mutex> lock (mutex.mutex);
+    const std::unique_lock<std::mutex> lock (mutex.mutex);
 
     // decrement the writer number first...
     --mutex.readerCount;
@@ -118,11 +118,11 @@ void MyWriterLock::acquire ()
         return;
     }
 
-    std::lock_guard<std::mutex> lock (mutex.mutex);
+    std::unique_lock<std::mutex> lock (mutex.mutex);
 
     // The writer count is not zero, so we have to wait for it to be zero again...
     while (mutex.writerCount != 0) {
-        mutex.cond.wait (mutex.mutex);
+        mutex.cond.wait (lock);
     }
 
     // ...then we can increment the writer count.
@@ -137,7 +137,7 @@ void MyWriterLock::release ()
         return;
     }
 
-    std::lock_guard<std::mutex> lock (mutex.mutex);
+    const std::unique_lock<std::mutex> lock (mutex.mutex);
 
     // Decrement the writer number first...
     if (--mutex.writerCount == 0) {
diff --git a/rtgui/threadutils.h b/rtgui/threadutils.h
index f54cbb845..9a7e810d8 100644
--- a/rtgui/threadutils.h
+++ b/rtgui/threadutils.h
@@ -91,7 +91,7 @@ public:
 
 private:
     std::mutex mutex;
-    std::condition_variable_any cond;
+    std::condition_variable cond;
 
     std::size_t writerCount = 0;
     std::size_t readerCount = 0;
```